testdata/p4_16_samples_outputs/gauntlet_hdr_assign_1-bmv2.p4-stderr

```diff
@@ -0,0 +1,6 @@
+gauntlet_hdr_assign_1-bmv2.p4(38): [--Wwarn=invalid_header] warning: accessing a field of an invalid header x
```

Comment:
I think this message is wrong. The previous line was `x = h.h`, so x.a may actually be valid if h.h.a was valid.

Reply:
Yes, I am aware of this issue. I commented today (above) about this situation, I think that the RemoveUnused pass removed that line.

> I forgot to mention the issue in the gauntlet_hdr_assign_1-bmv2.p4 test. If I am not wrong, the first FindUninitialized pass ends without warnings, then the RemoveUnused pass removes the assignment x = h.h in the ingress control, and finally, when these passes are repeated, we get some warnings, which I think should not be reported.

Here is the final output after all passes in simplifyDefUse finish:

    apply /* 
    <BlockStatement>(74255) */
    {
        /* 
      <MethodCallStatement>(62620)
        <MethodCallExpression>(62621)
          <Member>(62622)setInvalid
            <PathExpression>(62623)
              x_0/x
          <Vector<Type>>(61096), size=0
          <Vector<Argument>>(61094), size=0 */
        x_0.setInvalid();
        /* 
        <EmptyStatement>(74270) */    <-------------- this line
        ;
        /* 
      <AssignmentStatement>(62635)
        <Member>(62636)a
          <PathExpression>(62637)
            x_0/x
        <Constant>(1313) 2
          <Type_Bits>(289) */
        x_0.a = 16w2;
        /* 
      <AssignmentStatement>(62642)
        <Member>(46198)eth_type
          <Member>(46199)eth_hdr
            <PathExpression>(46200)
              h
        <Member>(62647)a
          <PathExpression>(62648)
            x_0/x */
        h.eth_hdr.eth_type = x_0.a;
    }

If you want, I can try to find out why this is happening.

Reply:
If that's what's happening it's clearly a bug, since x is not unused. Please file an issue about it.